src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpResponseStream.cs

```diff
@@ -12,26 +12,20 @@ namespace System.Net.Http
     internal class WinHttpResponseStream : Stream
     {
         private volatile bool _disposed = false;
-        private SafeWinHttpHandle _sessionHandle = null;
-        private SafeWinHttpHandle _connectHandle = null;
         private SafeWinHttpHandle _requestHandle = null;
 
-        internal WinHttpResponseStream(
-            SafeWinHttpHandle sessionHandle,
-            SafeWinHttpHandle connectHandle,
-            SafeWinHttpHandle requestHandle)
+        internal WinHttpResponseStream(SafeWinHttpHandle requestHandle)
         {
-            // While we only use the requestHandle to do actual reads of the response body,
-            // we need to keep the parent handles (connection, session) alive as well.
             bool ignore = false;
-            sessionHandle.DangerousAddRef(ref ignore);
-            connectHandle.DangerousAddRef(ref ignore);
             requestHandle.DangerousAddRef(ref ignore);
-            _sessionHandle = sessionHandle;
-            _connectHandle = connectHandle;
             _requestHandle = requestHandle;
         }
 
+        ~WinHttpResponseStream()
+        {
+            Dispose(false);
```

Comment:
I'm curious, why is the DangerousAddRef/DangerousRelease on _requestHandle necessary? (And hence why is the finalizer necessary?)  I understand that we want the request handle to be alive as long as the stream is alive.  Since WinHttpResponseStream has a reference to _requestHandle, the only way that _requestHandle would get ReleaseHandle'd would be if a) all references to WinHttpResponseStream were dropped such that the handle could be finalized, or b) some code elsewhere called Dispose on the handle while its ref count was zero.  The latter shouldn't happen, as these handles aren't expose out to arbitrary code, right?  And the former shouldn't matter, because if no references exist to the stream, no one will be trying to use it.  Worst case, if the underlying handle did get exposed, the Read calls that pass the SafeHandle to a P/Invoke will throw an exception due to trying to use a disposed object, but nothing should get corrupted.  Am I not understanding an important case?  I'm pushing on this simply because it'd be nice to avoid having these extra finalizers, if possible.  If it's necessary for some reason I've missed, looks good.